eng/scripts/Create-APIView.ps1

```diff
@@ -86,6 +86,32 @@ function Get-ResourceProviderFromReadMePath {
     return $null
 }
 
+function Get-ImpactedTypespecProjects {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$TypeSpecFile
+    )
+    $filePathParts = $TypeSpecFile.split([IO.Path]::DirectorySeparatorChar)
+    while ($filePathParts.Length -and !$configFilesInTypeSpecProjects) {
+      $filePathParts = $filePathParts | Select-Object -SkipLast 1
+      $typeSpecProjectBaseDirectory = $filePathParts -join [IO.Path]::DirectorySeparatorChar
```

Comment:
[Split-Path](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/split-path?view=powershell-7.4) makes pulling off the parents pretty simple. `Split-Path -Parent`